comps/agent/langchain/src/utils.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+
+from .config import env_config
+
+
+def format_date(date):
+    # input m/dd/yyyy hr:min
+    # output yyyy-mm-dd
+    date = date.split(" ")[0]  # remove hr:min
+    # print(date)
+    try:
+        date = date.split("/")  # list
+        # print(date)
+        year = date[2]
+        month = date[0]
+        if len(month) == 1:
+            month = "0" + month
+        day = date[1]
+        return f"{year}-{month}-{day}"
+    except:
+        return date
+
+
+def setup_hf_tgi_client(args):
+    from langchain_huggingface import HuggingFaceEndpoint
+
+    generation_params = {
+        "max_new_tokens": args.max_new_tokens,
+        "top_k": args.top_k,
+        "top_p": args.top_p,
+        "temperature": args.temperature,
+        "repetition_penalty": args.repetition_penalty,
+        "return_full_text": args.return_full_text,
+        "streaming": args.streaming,
+    }
```

Comment:
@XuhuiRen , updated, now use args for HuggingfaceEndpoint parameters configuration